checks/evaluation/signed_releases.go

```diff
@@ -17,118 +17,137 @@ package evaluation
+				Text: fmt.Sprintf("GitHub release found: %s", releaseName),
+			})
+			loggedReleases = append(loggedReleases, releaseName)
 		}
 
-		dl.Debug(&checker.LogMessage{
-			Text: fmt.Sprintf("GitHub release found: %s", release.TagName),
-		})
-
-		totalReleases++
-		signed := false
-		hasProvenance := false
-
-		// Check for provenance.
-		for _, asset := range release.Assets {
-			for _, suffix := range provenanceExtensions {
-				if strings.HasSuffix(asset.Name, suffix) {
-					dl.Info(&checker.LogMessage{
-						Path: asset.URL,
-						Type: finding.FileTypeURL,
-						Text: fmt.Sprintf("provenance for release artifact: %s", asset.Name),
-					})
-					hasProvenance = true
-					total++
-					break
-				}
-			}
-			if hasProvenance {
-				// Assign maximum points.
-				score += 10
-				break
-			}
+		// Check if outcome is NotApplicable
+		if f.Outcome == finding.OutcomeNotApplicable {
+			dl.Warn(&checker.LogMessage{
+				Text: "no GitHub releases found",
+			})
+			// Generic summary.
+			return checker.CreateInconclusiveResult(name, "no releases found")
+		}
+	}
+
+	totalPositive := 0
+	releaseMap := make(map[string]int)
+	uniqueReleaseTags := make([]string, 0)
+	checker.LogFindings(findings, dl)
+
+	for i := range findings {
+		f := &findings[i]
+
+		releaseName, err := getReleaseName(f)
+		if err != nil {
+			return checker.CreateRuntimeErrorResult(name, err)
 		}
 
-		if hasProvenance {
-			continue
+		if !contains(uniqueReleaseTags, releaseName) {
+			uniqueReleaseTags = append(uniqueReleaseTags, releaseName)
 		}
 
-		dl.Warn(&checker.LogMessage{
-			Path: release.URL,
-			Type: finding.FileTypeURL,
-			Text: fmt.Sprintf("release artifact %s does not have provenance", release.TagName),
-		})
-
-		// No provenance. Try signatures.
-		for _, asset := range release.Assets {
-			for _, suffix := range signatureExtensions {
-				if strings.HasSuffix(asset.Name, suffix) {
-					dl.Info(&checker.LogMessage{
-						Path: asset.URL,
-						Type: finding.FileTypeURL,
-						Text: fmt.Sprintf("signed release artifact: %s", asset.Name),
-					})
-					signed = true
-					total++
-					break
+		if f.Outcome == finding.OutcomePositive {
+			totalPositive++
+
+			switch f.Probe {
+			case releasesAreSigned.Probe:
+				if _, ok := releaseMap[releaseName]; !ok {
+					releaseMap[releaseName] = 8
 				}
-			}
-			if signed {
-				// Assign 8 points.
-				score += 8
-				break
+			case releasesHaveProvenance.Probe:
+				releaseMap[releaseName] = 10
 			}
 		}
+	}
 
-		if !signed {
-			dl.Warn(&checker.LogMessage{
-				Path: release.URL,
-				Type: finding.FileTypeURL,
-				Text: fmt.Sprintf("release artifact %s not signed", release.TagName),
-			})
-		}
-		if totalReleases >= releaseLookBack {
-			break
-		}
+	if totalPositive == 0 {
+		return checker.CreateMinScoreResult(name, "Project has not signed or included provenance with any releases.")
 	}
 
+	totalReleases := len(uniqueReleaseTags)
+
+	if totalReleases > 5 {
+		totalReleases = 5
+	}
```

Comment:
there's a subtle bug here I think.

Consider 10 releases, which all are signed, Score is 80. We cap num releases at 5,  and later on 
calc the score as 
```
score = int(math.Floor(float64(score) / float64(totalReleases)))
```

which would be 16, an impossible score.

Reply:
I see the probes limit it to 5, so no bug here. But later we may want to fix where the limit happens.